pkg/audit/disqualification_test.go

```diff
@@ -103,3 +107,111 @@ func calcReputation(dossier *overlay.NodeDossier) float64 {
+	// - mark a node as disqualified
+	// - check we don't get it when we require order limit
+
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+		upl := planet.Uplinks[0]
+
+		err := satellite.Audit.Service.Close()
+		require.NoError(t, err)
+
+		testData := make([]byte, 8*memory.KiB)
+		_, err = rand.Read(testData)
+		require.NoError(t, err)
+
+		err = upl.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		projects, err := satellite.DB.Console().Projects().GetAll(ctx)
+		require.NoError(t, err)
+		require.Len(t, projects, 1)
+
+		bucketID := []byte(storj.JoinPaths(projects[0].ID.String(), "testbucket"))
+
+		encScheme := upl.GetConfig(satellite).GetEncryptionScheme()
+		cipher := encScheme.Cipher
+		encryptedAfterBucket, err := streams.EncryptAfterBucket(ctx, "testbucket/test/path", cipher, &storj.Key{})
+		require.NoError(t, err)
+
+		lastSegPath := storj.JoinPaths(projects[0].ID.String(), "l", encryptedAfterBucket)
+		pointer, err := satellite.Metainfo.Service.Get(ctx, lastSegPath)
+		require.NoError(t, err)
+
+		disqualifiedNode := pointer.GetRemote().GetRemotePieces()[0].NodeId
+		disqualifyNode(t, ctx, satellite, disqualifiedNode)
+
+		limits, err := satellite.Orders.Service.CreateGetOrderLimits(ctx, upl.Identity.PeerIdentity(), bucketID, pointer)
+		require.NoError(t, err)
+
+		for _, orderLimit := range limits {
+			reputable, err := satellite.Overlay.Service.IsVetted(ctx, orderLimit.Limit.StorageNodeId)
+			require.NoError(t, err)
+			require.True(t, reputable)
+			require.NotEqual(t, orderLimit.Limit.StorageNodeId, disqualifiedNode)
+		}
+	})
+}
+
+func TestDisqualifiedNodesGetNoUpload(t *testing.T) {
+
+	// - mark a node as disqualified
+	// - check that we have an error if we try to create a segment using all storage nodes
+
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+		disqualifiedNode := planet.StorageNodes[0]
+
+		err := satellite.Audit.Service.Close()
+		require.NoError(t, err)
+
+		disqualifyNode(t, ctx, satellite, disqualifiedNode.ID())
+
+		request := overlay.FindStorageNodesRequest{
+			MinimumRequiredNodes: 4,
+			RequestedCount:       0,
+			FreeBandwidth:        0,
+			FreeDisk:             0,
+			ExcludedNodes:        nil,
+			MinimumVersion:       "", // semver or empty
+		}
+		nodes, err := satellite.Overlay.Service.FindStorageNodes(ctx, request)
+		require.True(t, overlay.ErrNotEnoughNodes.Has(err))
+
+		require.NotEmpty(t, nodes)
+		for _, node := range nodes {
+			reputable, err := satellite.Overlay.Service.IsVetted(ctx, node.Id)
+			require.NoError(t, err)
+			require.True(t, reputable)
+			require.NotEqual(t, node.Id, disqualifiedNode)
```

Comment:
Let's use `assert` instead of `require` for the above two checks.

We usually use `require` when there is no sense to continue the check if the assertion does not pass, e.g. when checking errors to avoid panics on the next checks. Here, if the test fails, we may have additional info from checking all nodes.